src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1711,23 +1711,24 @@ multi_join_restriction_hook(PlannerInfo *root,
 	MemoryContext restrictionsMemoryContext = plannerRestrictionContext->memoryContext;
 	MemoryContext oldMemoryContext = MemoryContextSwitchTo(restrictionsMemoryContext);
 
-	/*
-	 * We create a copy of restrictInfoList because it may be created in a memory
-	 * context which will be deleted when we still need it, thus we create a copy
-	 * of it in our memory context.
-	 */
-	List *restrictInfoList = copyObject(extra->restrictlist);
-
 	JoinRestrictionContext *joinRestrictionContext =
 		plannerRestrictionContext->joinRestrictionContext;
 	Assert(joinRestrictionContext != NULL);
 
+	Assert(!bms_is_empty(innerrel->relids) && !bms_is_empty(outerrel->relids));
```

Comment:
I think we are more and more  leaning towards not having  asserts, but instead have explicit error messages. The idea is that if we overlook something, in the production systems having the error messages much nicer compared to crashes.

For this case, I'd suggest we do something like the following:

```
if (bms_is_empty(innerrel->relids) || bms_is_empty(outerrel->relids))
{
   ereport(DEBUG1, "Join restriction information is NULL ");

   /* we cannot process empty relids, so ignoring the JoinRestriction is preferable over continuing and likely to  segfaulting  */
   return;
}
```

This check should go to the top of the function, definitely before doing `MemoryContextSwitchTo ()`

Reply:
or should we give an ERROR with ERRCODE_INTERNAL_ERROR instead of a debug message ?:

```c
		ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR),
				errmsg("<errorMessage>")));
```

Reply:
>ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR),
				errmsg("<errorMessage>")));

I  think, to be on  the  safe  side, we should probably not throw an error as those hooks are called for local tables. I see your point, we might return wrong results if we don't add (though we used to crash anyway). Still, I have a slight preference for DEBUG over ERROR